lib/Sema/NameBinding.cpp

```diff
@@ -297,6 +298,551 @@ static void insertPrecedenceGroupDecl(NameBinder &binder, SourceFile &SF,
+        if (!config) {
+          Elements.push_back(transformStmt(stmt));
+          continue;
+        }
+
+        // Now that we have the active clause, look into its elements for
+        // further statements to transform.
+        for (auto &clause : config->getClauses()) {
+          // Evaluate conditions until we find the active clause.
+          if (clause.Cond && !evaluateCondition(clause.Cond)) {
+            continue;
+          }
+
+          for (auto &clauseElt : clause.Elements) {
+            // Now that we have the active clause, look into its members for
+            // further decls.
+            if (auto TLD = clauseElt.dyn_cast<Decl *>()) {
+              // Nested top-level code declarations are a special case that
+              // require extra transformation.
+              if (auto *TLCD = dyn_cast<TopLevelCodeDecl>(TLD)) {
+                if (TLCD->isImplicit()) {
+                  continue;
+                }
+
+                if (BraceStmt *Body = TLCD->getBody()) {
+                  BraceStmt *NewBody = resolveClausesAndExtractTLCDs(Body);
+                  if (NewBody != Body) {
+                    TLCD->setBody(NewBody);
+                  }
+                }
+                ExtraTLCDs.push_back(TLCD);
+              } else {
+                // For newly-inserted members of aggregates and extensions we
+                // need to warn if they're marked '@objc'.
+                if (auto *IDC = dyn_cast<IterableDeclContext>(TLD)) {
+                  for (auto member : IDC->getMembers()) {
+                    for (auto Attr : member->getAttrs()) {
+                      if (isa<ObjCAttr>(Attr) || isa<DynamicAttr>(Attr))
+                        Binder.SF.AttrsRequiringFoundation
+                          .insert({Attr->getKind(), Attr});
+                    }
+                  }
+                }
+                ExtraTLCDs.push_back(TLD);
+              }
+            }
+          }
+          break;
+        }
+
+      }
+      return swift::BraceStmt::create(Context, BS->getLBraceLoc(),
+                                      Context.AllocateCopy(Elements),
+                                      BS->getRBraceLoc());
+    }
+
+    virtual bool walkToDeclPre(Decl *D) {
+      if (TopLevelCodeDecl *TLCD = dyn_cast<TopLevelCodeDecl>(D)) {
+        // For top-level declarations in libraries or in the interpreter we
+        // extract and evaluate any conditions and their corresponding blocks.
+        if (TLCD->isImplicit()) {
+          return false;
+        }
+
+        if (BraceStmt *Body = TLCD->getBody()) {
+          BraceStmt *NewBody = resolveClausesAndExtractTLCDs(Body);
+          if (NewBody != Body) {
+            TLCD->setBody(NewBody);
+          }
+        }
+      } else if (AbstractFunctionDecl *FD = dyn_cast<AbstractFunctionDecl>(D)) {
+        // For functions, transform the statements in the body.
+        // FIXME: This can be deferred until type checking.
```

Comment:
Yeah, I think we do need to figure out how to do this. Code completion doesn't parse function bodies unless it needs to, but this will probably fault them all in eagerly.